### DISCUSSION TITLE
Making a GeometryCollection an iterable object

### DISCUSSION THREAD

**Discussion by swiss-knight** [shapely/shapely#1750] [Q&A] [answered]
Hello,

would it be possible to make a GeometryCollection object iterable?

Per the [documentation](https://shapely.readthedocs.io/en/latest/reference/shapely.GeometryCollection.html):

```
from shapely import LineString, Point
p = Point(51, -1)
l = LineString([(52, -1), (49, 2)])
gc = GeometryCollection([p, l])
```

but:

```
list(gc)

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'GeometryCollection' object is not iterable
```

It was actually iterable in shapely `1.8.5`.

Sorry if I missed something obvious, but I dug into the methods of the class without any success for the moment.

Thanks.

**Comment by jorisvandenbossche** [comment, marked as answer]
This was indeed changed in in the Shapely 1.8 -> 2.0 transition. You can read here about this here: https://shapely.readthedocs.io/en/latest/migration.html#multi-part-geometries-will-no-longer-be-sequences-length-iterable-indexable (the main reason for this is to make it more robust to work with arrays of geometries. The section about this in the RFC has some more details on that [here](https://github.com/shapely/shapely-rfc/blob/df79554e19aa7182796eaa3fed22e0e7084607ef/rfc/0001-shapely-2.0-roadmap.md#iteration--getitem-of-multi-geometries))

So given that we intentionally changed this, there are no plans to make it iterable again. You can use the `.geoms` attribute instead:

```
>>> list(gc.geoms)
[<POINT (51 -1)>, <LINESTRING (52 -1, 49 2)>]
```

We could maybe provide a custom error message when trying to iterate that can explicitly point users towards `.geoms`. We could also update the documentation you linked to to highlight that, while it represents a "collection", the object itself is not considered a Collection in Python's ABC terms.

**Reply by swiss-knight** [reply]
Thanks a lot for the very fast reply, this saved me a lot of troubles!

I hope it will help many other users of this great geo-library! 😊
